lib/src/boilerplate_suggestors/boilerplate_utilities.dart

```diff
@@ -211,12 +253,44 @@ class ClassToMixinConverter {
       if (shouldSwapParentClass) {
         yieldPatch(
             node.extendsClause.superclass.name.offset,
-            node.extendsClause.superclass.name.end,
+            node.extendsClause.superclass.end,
             isAPropsClass(node) ? 'UiProps' : 'UiState');
       }
 
-      if (node.withClause != null) {
-        yieldPatch(node.withClause.offset, node.withClause.end, '');
+      final shouldAddImplementsClause =
+          node.withClause != null || shouldSwapParentClass;
+      if (shouldAddImplementsClause) {
+        // Add an implements clause since the mixin is how consumers will "extend" props moving forward.
+        // These mixins are moved to the new concrete class via the `AdvancedPropsAndStateClassMigrator`,
+        // but if they are not also implemented here, the new props mixin won't have all
+        // the keys that the old class / abstract class had... which will be a breaking
+        // change for consumers without a workaround/path forward.
+        final mixinTypeNames = shouldSwapParentClass
+            ? [node.extendsClause.superclass, ...?node.withClause?.mixinTypes]
+            : node.withClause?.mixinTypes;
+        final mixinNames =
+            mixinTypeNames.joinByName(converter: this, sourceFile: sourceFile);
+
+        if (node.withClause != null) {
+          if (node.implementsClause != null) {
+            // `with` and `implements` clause exist
+            yieldPatch(node.withClause.offset, node.withClause.end, '');
+            yieldPatch(node.implementsClause.end, node.implementsClause.end,
+                ', $mixinNames');
+          } else {
+            // only `with` clause exists
+            yieldPatch(node.withClause.offset, node.withClause.end,
+                ' implements $mixinNames');
+          }
+        } else if (node.implementsClause != null) {
+          // only `implements` clause exists
+          yieldPatch(node.implementsClause.end, node.implementsClause.end,
+              ', $mixinNames');
+        } else {
+          // no `with` or `implements` clause exist
+          yieldPatch(node.leftBracket.offset - 1, node.leftBracket.offset - 1,
```

Comment:
It's not safe to assume that there's a whitespace before the left bracket. This should just be
```suggestion
          yieldPatch(node.leftBracket.offset, node.leftBracket.offset,
```